plugins/introduction.rst

```diff
@@ -47,43 +46,48 @@ Let’s see an example::
         ]
     );
 
-The Plugin Client accepts and implements both ``Http\Client\HttpClient`` and ``Http\Client\HttpAsyncClient``, so you can use
-both ways to send a request. In case the passed client implements only one of these interfaces, the Plugin Client
+The ``PluginClient`` accepts and implements both ``Http\Client\HttpClient`` and
+``Http\Client\HttpAsyncClient``, so you can use both ways to send a request. In
+case the passed client implements only one of these interfaces, the ``PluginClient``
 "emulates" the other behavior as a fallback.
 
-It is important, that the order of plugins matters. During the request, plugins are called in the order they have
-been added, from first to last. Once a response has been received, they are called again in reversed order,
-from last to first.
+It is important to note that the order of plugins matters. During the request,
+plugins are executed in the order they have been specified in the constructor,
+from first to last. Once a response has been received, the plugins are called
+again in reversed order, from last to first.
+
+For our previous example, the execution chain will look like this:
 
-In case of our previous example, the execution chain will look like this::
+.. code::
 
     Request  ---> PluginClient ---> RetryPlugin ---> RedirectPlugin ---> HttpClient ----
                                                                                        | (processing call)
     Response <--- PluginClient <--- RetryPlugin <--- RedirectPlugin <--- HttpClient <---
 
-In order to have correct behavior over the global process, you need to understand well how each plugin is used,
-and manage a correct order when passing the array to the Plugin Client.
+In order to achieve the intended behavior in the global process, you need to
+pay attention to what each plugin does and define the correct order accordingly.
 
-Retry Plugin will be best at the end to optimize the retry process, but it can also be good
-to have it as the first plugin, if one of the plugins is inconsistent and may need a retry.
+For example, the ``RetryPlugin`` should probably be at the end of the chain to
+keep the retry process as short as possible. However, if one of the other
+plugins is doing a fragile operation that might need a retry, place the retry
+plugin before that.
 
 The recommended way to order plugins is the following:
 
- 1. Plugins that modify the request should be at the beginning (like Authentication or Cookie Plugin)
- 2. Plugins which intervene in the workflow should be in the "middle" (like Retry or Redirect Plugin)
- 3. Plugins which log information should be last (like Logger or History Plugin)
+ 1. Plugins that modify the request should be at the beginning (like Authentication or Cookie Plugin);
+ 2. Plugins which intervene in the workflow should be in the "middle" (like Retry or Redirect Plugin);
+ 3. Plugins which log information should be last (like Logger or History Plugin).
```

Comment:
this is the symfony doc style afaik. i can't find it mentioned explicitly on http://symfony.com/doc/current/contributing/documentation/standards.html#code-examples but at least that example also has them.

but i am fine with removing them again if others also don't like them. i got used to them, so i'd vote to keep doing them, but its no big deal for me.